test/test_helpers.py

```diff
@@ -118,6 +118,43 @@ def test_some_headers(self):
         assert examples.dataset.headers == ["im", ""]
 
 
+def test_example_caching_relaunch(connect):
+    def combine(a, b):
+        return a + " " + b
+
+    with gr.Blocks() as demo:
+        txt = gr.Textbox(label="Input")
+        txt_2 = gr.Textbox(label="Input 2")
+        txt_3 = gr.Textbox(value="", label="Output")
+        btn = gr.Button(value="Submit")
+        btn.click(combine, inputs=[txt, txt_2], outputs=[txt_3])
+        gr.Examples(
+            [["hi", "Adam"], ["hello", "Eve"]],
+            [txt, txt_2],
+            txt_3,
+            combine,
+            cache_examples=True,
+            api_name="examples",
+        )
+
+    with connect(demo, server_port=7859) as client:
+        assert client.predict(1, api_name="/examples") == (
+            "hello",
+            "Eve",
+            "hello Eve",
+        )
+
+    # Let the server shut down
+    time.sleep(1)
+
+    with connect(demo, server_port=7859) as client:
```

Comment:
Just to confirm, you launch on the same port to ensure that the server truly has shutdown? I don't think you need to do this.

Even if `demo.launch()` were to launch a FastAPI app on a new port, it would still read from the same cache since the ids of the components hasn't changed.

So you can remove the hardcoding of the server ports (which will be better in case this port is occupied)

Reply:
The time.sleep is to ensure shutdown. Hardcoding the same port is for paranoia 😂 Let me fix.